### ISSUE TITLE
Notifications: Jetpack Monitor link interception issue

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-iOS#8188]
### Expected behavior

I expect that tapping the link in "Learn more about Jetpack Monitor" in this notification will take me to the page https://jetpack.com/support/monitor/ on the Jetpack site:

![2017-11-09-16-14-20](https://user-images.githubusercontent.com/8658164/32959647-2d339810-cbba-11e7-9fb5-92e9b1037512.png)

### Actual behavior

The link opens this Reader post:

![2017-11-09-16-14-24](https://user-images.githubusercontent.com/8658164/32959662-3a928390-cbba-11e7-97da-9e125caa41d2.png)

### Steps to reproduce the behavior

1. Create a test Jetpack site and connect it to WordPress.com.
2. Enable Jetpack Monitor on the site.
3. Enable WordPress.com notifications for Monitor in the Jetpack site’s [security settings](https://wordpress.com/settings/security).
4. Take the site down. (This should trigger a Jetpack Monitor notification.)
5. Open the app and go to the notifications tab.
6. Open the Jetpack Monitor notification and tap the link to learn more.

##### Tested on iPhone 6, iOS 11.1.1, WPiOS 8.8.0.20171116

**Comment by elibud** [comment]
@jleandroperez can you take a look at this one?

**Comment by jleandroperez** [comment]
### Update:
- Unable to trigger the wpcom notification (as per Jan 2, 2018)
- This kind of notification is locked down, for now, to a11ns

I'll be bugging the Jetpack team, it's likely it was either (completely) disabled, or there's something broken. The bug itself looks like a data format issue, which we can definitely improve.

--

@drw158 (hi Dave!!) do you know who would be the design expert maintaining Notifications these days? (this one could probably be rebuilt as a "Badge" notification for iOS / Android / Web).

Thanks in advance!!

**Comment by jleandroperez** [comment]
Ref.: [Jetpack Issue Mark 7892](https://github.com/Automattic/jetpack/issues/7892)

**Comment by jleandroperez** [comment]
### Update Mark II:

(@enejb Thanks a LOOOT for helping me debug this one!!). We've tracked down this problem to the Ranges payload, for the `Learn More about Jetpack Monitor` link.

TL;DR: In order to push a WebView, we need to remove the fields `id` + `site_id` + `type` from the range that carries the link.

@dmsnell can we bug you a little bit with this one? we've been checking out the Notification's Builder `build_jetpack_monitor_note` method, and turns out... the ID is not being explicitly set anywhere.

Thanks in advance!!

```
{   
    ranges = (
        {
            id = 9;
            indices = (18, 33);
            "site_id" = 20115252;
            type = post;
            url = "https://jetpack.com/support/monitor/";
        },
        {
            indices = (0, 0 );
            type = noticon;
            value = "\Uf455";
        }
    );
    text = " Learn more about Jetpack Monitor.";
}
```

**Comment by dmsnell** [comment]
@jleandroperez why do those properties need to go away? shouldn't they just be ignored and extra info?

**Comment by jleandroperez** [comment]
@dmsnell reason is: because of duck typing, both Android and iOS end up pushing the native reader, which results in a blank screen. 

I'll try to debug the reader (not sure what's causing the blank screen // Eric is on sabattical), but still, since this issue affects both, Android *and* iOS, it's probably a great idea to force a WebView here.

**Comment by drw158** [comment]
No designer on Notifications right now, just me.

**Comment by jleandroperez** [comment]
Fixed by @dmsnell via Trac 167725.

Thaaaaanks Dennis!!
